plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/ast/Callable.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.viper.ast
+
+import scala.collection.JavaConverters
+
+/**
+ * Interface used to wrap Viper nodes with an argument list (such as function applications, method calls, ...).
+ */
+interface IntoCallable {
+    fun asCallable(): Callable
+}
+
+/**
+ * Represents a Viper-callable object with arguments.
+ */
+interface Callable {
+    val args: List<Node>
+
+    fun arg(index: Int): Node {
+        val arguments = args
+        require(index in arguments.indices) {
+            "The callable does not have a ${index + 1}-th argument."
+        }
+        return arguments[index]
+    }
+
+    /**
+     * Extract Info metadata from the i-th argument list.
+     */
+    fun argInfo(index: Int): Info = arg(index).info
```

Comment:
Wouldn't it make more sense for a `Callable` to provide access to its arguments as `Node`s (or `Exp`s) and then allow one to get the info via that interface?